### ISSUE TITLE
call for new feature similar to function __clz() in cuda

### ISSUE THREAD

**Issue by Yihao-Shi** [taichi-dev/taichi#8212]
In the process of building linear bvh in gpu, a good idea is that dividing the objectsd by the highest differing bit in their Morton codes, corresponds to classifying them on either side of an axis-aligned plane in 3D. Thus, cuda provides intrinsic function __clz() to count the number of leading zero bits in a 32-bit integer. However, Taichi is not aviliable for this feature (but it is important in building linear bvh).
I am calling for adding the features which is similar to __clz().

**Comment by lin-hitonami** [comment]
We welcome contribution for this feature. We are happy to offer help if anyone is interested. [This PR](https://github.com/taichi-dev/taichi/pull/7772) which adds popcnt to Taichi may be helpful on how to add intrinsics to Taichi.

**Comment by JettChenT** [comment]
Hi! I'm willing to work on this issue.

**Comment by lin-hitonami** [comment]
> Hi! I'm willing to work on this issue.

Thank you! Please let us know if you need any assistance.

**Comment by JettChenT** [comment]
Thanks! To be clear, I am writing a manual implementation for `__clz()` feature for cuda with python similar to the mentioned PR, eg..
```python
for i in range(32):
    if 2**i > n:
        return 32 - i
```
, right?

**Comment by lin-hitonami** [comment]
Not exactly. We should add an intrinsic to the IR and use the built-in intrinsics in the LLVM and SPIRV based backends.
We can use `llvm::Intrinsic::ctlz` in LLVM and use [`32 - findMSB`](https://registry.khronos.org/OpenGL-Refpages/gl4/html/findMSB.xhtml) in SPIRV backend.

**Comment by JettChenT** [comment]
noted, thanks!
